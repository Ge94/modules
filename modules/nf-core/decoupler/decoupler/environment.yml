---
# yaml-language-server: $schema=https://raw.githubusercontent.com/nf-core/modules/master/modules/environment-schema.json
channels:
  - conda-forge
  - bioconda
dependencies:
  # renovate: datasource=conda depName=conda-forge/decoupler-py
  - conda-forge::decoupler-py=1.6.0
