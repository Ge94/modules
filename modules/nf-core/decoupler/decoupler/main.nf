process DECOUPLER {
    tag "${meta.id}"
    label 'process_medium'
    conda "${moduleDir}/environment.yml"
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/cf/cf3578c5f4a5ad40d10b535d496daa29f47e908fbbbc789f85e3341b8f9341ba/data'
        : 'community.wave.seqera.io/library/decoupler-py:1.8.0--b139c6574c0ac2a6'}"

    input:
    tuple val(meta), path(mat)
    path net

    output:
    tuple val(meta), path("*estimate__decoupler.tsv"), emit: dc_estimate
    tuple val(meta), path("*pvals__decoupler.tsv"), emit: dc_pvals
    path ("versions.yml"), emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: "{}"
    """
    #!/usr/bin/env python3
    import os
    import pandas as pd

    os.environ["NUMBA_CACHE_DIR"] = "./tmp"

    import decoupler as dc

    methods = ['aucell', 'gsea', 'gsva', 'mdt', 'mlm', 'ora', 'udt',
        'ulm', 'viper', 'wmean', 'wsum']

    mat = pd.read_csv("${mat}", sep="\t", index_col=0)
    net = pd.read_csv("${net}", sep="\t", index_col=0)

    # Parsing arguments
    args = ${args}
    parsedargs = {'args': {}}

    for k, v in args.items():
        # Specific method argument
        if k.split('_')[0] in methods:
            meth = k.split('_')[0]
            arg = '_'.join(k.split('_')[1:])

            if meth not in args['args'].keys():
                parsedargs['args'][meth] = {arg: v}
            else:
                parsedargs['args'][meth].update({arg: v})

        # Generic argument
        else:
            parsedargs[k] = v


    results = dc.decouple(
        mat=mat,
        net=net,
        **parsedargs
    )

    for result in results:
        results[result].to_csv(result + "__decoupler.tsv", sep="\t")

    ## VERSIONS FILE
    with open('versions.yml', 'a') as version_file:
        version_file.write('"${task.process}":' + "\\n")
        version_file.write("\tdecoupler-py: " + dc.__version__ + "\\n")
    """
}
