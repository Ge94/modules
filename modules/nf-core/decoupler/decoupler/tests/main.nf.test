nextflow_process {

    name "Test Process DECOUPLER"
    script "../main.nf"
    config "./nextflow.config"
    process "DECOUPLER"

    tag "modules"
    tag "modules_nfcore"
    tag "decoupler"
    tag "decoupler/decoupler"

    test("decoupler_test") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'generic/tsv/expression.tsv', checkIfExists: true)
                ]
                input[1] = [
                    file(params.modules_testdata_base_path + 'generic/tsv/network.tsv', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert snapshot(path(process.out.versions.get(0)).yaml).match("versions") },
            )
        }

    }

    test("decoupler_test - stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'generic/tsv/expression.tsv', checkIfExists: true)
                ]
                input[1] = [
                    file(params.modules_testdata_base_path + 'generic/tsv/network.tsv', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}