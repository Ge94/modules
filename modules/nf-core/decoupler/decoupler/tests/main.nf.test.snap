{
    "versions": {
        "content": [
            {
                "DECOUPLER": {
                    "decoupler-py": "1.8.0"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-22T15:53:36.673249"
    },
    "decoupler_test": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "consensus_estimate__decoupler.tsv:md5,3adfc527d7491baaec5a9ba003345b06",
                            "mlm_estimate__decoupler.tsv:md5,10f0a0a82dc252e2843df04430daa30c",
                            "ulm_estimate__decoupler.tsv:md5,3f38c1f33d6fdf352bc311e269662c79",
                            "wsum_estimate__decoupler.tsv:md5,5e6c68334b105b341042507638e3a950"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "consensus_pvals__decoupler.tsv:md5,01b956e8cd0c0f24858c83382792f3ba",
                            "mlm_pvals__decoupler.tsv:md5,70e53ea27d51f0b74de08e09cfa0d22a",
                            "ulm_pvals__decoupler.tsv:md5,024d29e68941f56d4f2e878826ad4ded",
                            "wsum_pvals__decoupler.tsv:md5,fac9e18b9495ca8a9ae92a015eed3043"
                        ]
                    ]
                ],
                "2": [
                    "versions.yml:md5,8fd115b4099050429b348981ca1d5e86"
                ],
                "dc_estimate": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "consensus_estimate__decoupler.tsv:md5,3adfc527d7491baaec5a9ba003345b06",
                            "mlm_estimate__decoupler.tsv:md5,10f0a0a82dc252e2843df04430daa30c",
                            "ulm_estimate__decoupler.tsv:md5,3f38c1f33d6fdf352bc311e269662c79",
                            "wsum_estimate__decoupler.tsv:md5,5e6c68334b105b341042507638e3a950"
                        ]
                    ]
                ],
                "dc_pvals": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "consensus_pvals__decoupler.tsv:md5,01b956e8cd0c0f24858c83382792f3ba",
                            "mlm_pvals__decoupler.tsv:md5,70e53ea27d51f0b74de08e09cfa0d22a",
                            "ulm_pvals__decoupler.tsv:md5,024d29e68941f56d4f2e878826ad4ded",
                            "wsum_pvals__decoupler.tsv:md5,fac9e18b9495ca8a9ae92a015eed3043"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8fd115b4099050429b348981ca1d5e86"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-22T15:53:36.587951"
    }
}